### PR TITLE
feat: adapt "FileSeries" with both "RemoteFile" and "File"

### PR DESCRIPTION
**Pull request by linjiX**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2a8n66f 统一 RemoteFile 和 File 的接口](https://app.clickup.com/t/2a8n66f)

File: graviti/utility/file.py [hunk 1]
```diff
@@ -43,6 +56,25 @@ def checksum(self) -> str:
...
+        url: str = scalar["url"].as_py()
+        checksum: str = scalar["checksum"].as_py()
+
+        if url.startswith("file://"):
+            return File._create(url, checksum)  # pylint: disable=protected-access
+
+        return RemoteFile(url, checksum)
+
 
 class RemoteFile(FileBase):
```
Comment (Lee-000) [inline graviti/utility/file.py:79]: Does `RemoteFile` need a `to_pyobj` method?

File: graviti/utility/file.py [hunk 2]
```diff
@@ -53,27 +85,24 @@ class RemoteFile(FileBase):
 
     """
 
-    __slots__ = ("url", "_checksum")
+    __slots__ = FileBase.__slots__
```
[the remark this answers is not in the document]
Comment (linjiX) [reply graviti/utility/file.py:88]: `__slots__` cannot be inherited

File: graviti/utility/file.py (hunk 1 above)
Comment (linjiX) [reply graviti/utility/file.py:79]: Is not necessary for now

File: graviti/utility/file.py [hunk 3]
```diff
@@ -84,14 +113,37 @@ class File(FileBase):
...
-        self.path = Path(path)
+        self.path = Path(path).absolute()
 
     def _repr_head(self) -> str:
         return f'{self.__class__.__name__}("{self.path.name}")'
 
+    @classmethod
+    def _create(cls: Type[_T], url: str, checksum: str) -> _T:
+        obj: _T = object.__new__(cls)
+        obj.path = Path(url[7:])
```
[the remark this answers is not in the document]
Comment (linjiX) [reply graviti/utility/file.py:129]: remove the "file://" in start of the url

File: graviti/utility/file.py [hunk 4]
```diff
@@ -84,14 +113,37 @@ class File(FileBase):
...
+        return obj
+
+    @property
+    def url(self) -> str:
+        """Get the sha1 checksum of the local file.
+
+        Returns:
+            The sha1 checksum of the local file.
+
+        """
```
Comment (Lee-000) [inline graviti/utility/file.py:141]: The docstring should be updated.

File: graviti/utility/file.py [hunk 5]
```diff
@@ -84,14 +113,37 @@ class File(FileBase):
...
+        self.path = Path(path).absolute()
 
     def _repr_head(self) -> str:
         return f'{self.__class__.__name__}("{self.path.name}")'
 
+    @classmethod
+    def _create(cls: Type[_T], url: str, checksum: str) -> _T:
+        obj: _T = object.__new__(cls)
+        obj.path = Path(url[7:])
+        obj._url = url
```
Comment (AChenQ) [inline graviti/utility/file.py:130]: We need add `_url` in `to_pyobj`， We need to save the `path` and `checksum` in pyarrow array.

File: graviti/utility/file.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (linjiX) [reply graviti/utility/file.py]: the base class `ReprMixin` has no `__slots__`, the `__slots__` can not block new attr set, remove the `__slots__` for now.

**Review by AChenQ — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**